File: src/java/org/javaturk/dp/ch03/ocp/factory/solution/Employee.java
package org.javaturk.dp.ch03.ocp.factory.solution;

public class Employee{
	protected int no;
	protected String name;
	protected int year;
	protected String department;
	protected double salary;
	
	public static final int BASE_SALARY = 500;
	
	public Employee(int no, String name, int year, String department){
		this.no = no;
		this.name = name;
		this.year = year;
		this.department = department;
	}
	
	public void work(){
		System.out.println("Employee is working!");
	}

	public double calculateSalary(){
		salary = year * BASE_SALARY;
		return salary;
	}
	
	public void printInfo(){
		System.out.println("\nEmployee Info");
		System.out.println("No: " + no);
		System.out.println("Name:" + name);
		System.out.println("Year: " + year);
		System.out.println("Department: " + department);
	}
	
	public int getNo() {
		return no;
	}

	public String getName() {
		return name;
	}

	public int getYear() {
		return year;
	}

	public String getDepartment() {
		return department;
	}

	public double getSalary() {
		return salary;
	}

	@Override
	public String toString() {
		return "Employee [no=" + no + ", name=" + name + ", year=" + year + ", department=" + department + ", salary="
				+ salary + "]";
	}
}

File: src/typescript/com/gokselkucuksahin/dp/creational-patterns/factory/employee1/employee.ts
export class Employee {
  protected id: number;
  protected name: string;
  protected year: number;
  protected department: string;
  protected salary: number;

  static readonly BASE_SALARY: number = 500;

  constructor(id: number, name: string, year: number, department: string) {
    this.id = id;
    this.name = name;
    this.year = year;
    this.department = department;
    this.calculateSalary();
  }

  work(): void {
    console.log("Employee is working!");
  }

  calculateSalary(): number {
    this.salary = this.year * Employee.BASE_SALARY;
    return this.salary;
  }

  printInfo(): void {
    console.log(`
Employee Info
Id: ${this.id}
Name: ${this.name}
Year: ${this.year}
Department: ${this.department}`);
  }

  getId(): number {
    return this.id;
  }

  getName(): string {
    return this.name;
  }

  getYear(): number {
    return this.year;
  }

  getDepartment(): string {
    return this.department;
  }

  getSalary(): number {
    return this.salary;
  }

  toString(): string {
    return `Employee [id=${this.id}, name=${this.name}, year=${this.year}, department=${this.department}, salary=${this.salary}]`;
  }
}